protocol/messenger.go

```diff
@@ -2917,6 +2927,7 @@ func (m *Messenger) handleRetrievedMessages(chatWithMessages map[transport.Filte
 					c, err := buildContact(publicKey)
 					if err != nil {
 						logger.Info("failed to build contact", zap.Error(err))
+						messageProcessed = false
 						continue
 					}
```

Comment:
Part of me thinks that we may need to wrap this failure functionality, future changes may forget to include the new `messageProcessed = false` line.

We could maintain the `messageProcessed` state in the Messenger struct and call an unexported func a something like:

```
handledMessageProcessFail(msg string, err error, l logger) {
  l.Info(msg, err)
  m.allMessagesProcessed = false
}
```

This could be over engineering. I think if we add more than just logging and setting a var to false 100% this should be wrapped, but at the moment we're sort of on the border.

Maybe leave it as it is but keep an eye on if we need to add more functionality to a failed message process.